File: Python/21-30/e_26.py
# Escribe un programa en python que muestre la carpeta en la que se encuentra el script actual


import os

# Obtener la ruta absoluta del directorio actual del script
ruta_actual = os.path.abspath(os.path.dirname(__file__))

print(f"El script se encuentra en la carpeta: {ruta_actual}")



# Puedes usar el módulo os en Python para obtener la carpeta en la que se encuentra el script actual.
# En este ejemplo, os.path.dirname(__file__) obtiene el directorio del script actual y os.path.abspath() obtiene la ruta absoluta de ese directorio. 
# Finalmente, se imprime la carpeta en la que se encuentra el script.
#

File: JS/21-30/e_26.js
// Obtener la ruta del script actual
const rutaActual = new URL(import.meta.url).pathname;

// Obtener la carpeta del script actual
const carpetaActual = new URL('.', rutaActual).pathname;

console.log(`El script se encuentra en la carpeta: ${carpetaActual}`);



// En JavaScript, puedes utilizar el objeto import.meta para obtener información sobre el módulo actual, incluida la ruta del archivo.
// Este código utiliza la API de URL para manipular la ruta del módulo actual y extraer la carpeta.
// Ten en cuenta que el uso de import.meta es compatible con los módulos ECMAScript, y este código funcionará en entornos que admitan esta funcionalidad,
// como en navegadores modernos o en entornos Node.js con soporte para módulos ECMAScript.